### PR TITLE
#283 - AssetDetails404Servlet should work w/ Sling error handling

### PR DESCRIPTION
**Pull request by kaushalmall**

### CODE REVIEW

File: core/src/main/java/com/adobe/aem/commons/assetshare/configuration/impl/AssetDetails404Servlet.java [hunk 1]
```diff
@@ -52,10 +52,7 @@
     protected final void doGet(SlingHttpServletRequest request, SlingHttpServletResponse response)
             throws ServletException, IOException {
         // Control which requests enter this method via accepts(..) below.
-
-        response.setStatus(SlingHttpServletResponse.SC_NOT_FOUND);
-        response.getWriter().print("The path to a valid and accessible asset must be provided to the Asset Details page for it to display.");
-        response.getWriter().flush();
+        response.sendError(SlingHttpServletResponse.SC_NOT_FOUND);
```
Comment (davidjgonzalez) [inline core/src/main/java/com/adobe/aem/commons/assetshare/configuration/impl/AssetDetails404Servlet.java]: Think its helpful to pass that message as param #2? I assume most ppl override the error page at some level, but maybe for "debugging"? Or better to just leave blank?

**Comment by davidjgonzalez** [comment]
Based on https://sling.apache.org/documentation/the-sling-engine/errorhandling.html#http-status-codes that looks like it should work - but would have to try

**Comment by kaushalmall** [comment]
@davidjgonzalez how about doing it the way of the latest commit? I know the ACS Commons version is not up to date, can change that.

File: core/pom.xml [hunk 2]
```diff
@@ -132,6 +132,10 @@
             <groupId>org.apache.commons</groupId>
             <artifactId>commons-email</artifactId>
         </dependency>
+		<dependency>
```
Comment (davidjgonzalez) [inline core/pom.xml]: I dont think we should depend on ACS Commmons... i guess this should really work with any Sling-based error handling?

**Comment by kaushalmall** [comment]
@davidjgonzalez not quite sure why this isn't working. Although, I do something weird when debugging ACS Commons EPH. 

When I access a URL an invalid URL where the ASC OptingServlet's accept method would return false, for example`/homes.html`. In EPH, the path is same as what I am requesting. 

image

But, when I access an invalid URL where the ASC OptingServlet's accept method would return true, for example `/content/microsoft/search-marketing/home/asset-details/image.html/content/dam/1.tiff?wcmmode=disabled`. In EPH, the path is not what I am requesting. 

image

Not quite sure what to do now....

**Comment by kaushalmall** [comment]
This is an old one. Invalid now, closing.

**Pull request closed without merging**